Atomic/objects/image.py

```diff
@@ -109,21 +110,24 @@ def propagate(_img):
             img = self.registry
             if self.repo:
                 img += "/{}".format(self.repo)
-            img += "/{}:{}".format(self.image, self.tag)
+            if self.tag:
+                img += "/{}:{}".format(self.image, self.tag)
+            elif self.digest:
+                img += "/{}@{}".format(self.image, self.digest)
             self._fq_name = img
             propagate(self._fq_name)
             return img
 
         if not self.registry:
             ri = RegistryInspect(registry=self.registry, repo=self.repo, image=self.image,
-                                 tag=self.tag, orig_input=self.input_name)
+                                 tag=self.tag, digest=self.digest, orig_input=self.input_name)
```

Comment:
Should tag and digest be combined like you did below? IE You pass either the tag or the digest to tagdigest field?